contrib/friendz/src/friendz.nit

```diff
@@ -920,28 +904,29 @@ redef class Game
 	# SOUND
 
 	# Is the music muted?
-	var music_muted: Bool = true #read_cookie("music_muted")
+	var music_muted: Bool = false #read_cookie("music_muted")
 
 	# Is the sound effects muted?
-	var sfx_muted: Bool = true #read_cookie("sfx_muted")
+	var sfx_muted: Bool = false #read_cookie("sfx_muted")
 
 	# The background music resource. */
-	var music = new Audio("music.ogg")
+	var music: Sound is writable
 
 	# Click sound
-	var snd_click = new Audio("click.wav")
+	var snd_click: Sound is writable
```

Comment:
Aucune idée.

`app` est global donc on pourait charger directement en causant à `app`.

~~~nit
redef class Game
   var x: Sound = app.load_sound("x.wav")
end
~~~

ou passer par un service de Game pour eviter de causer directement à `app`

~~~nit
redef class Game
   var x: Sound = load_sound("x.wav")
   fun load_sound(filename: String): Sound
   do
       do return app.load_sound(finename).as(not null)
   end
end
~~~

Une autre idée à considerer (par pour cette PR) est le chargement en deux temps. C'est peut-etre une bonne approche : 1. on cree un objet qui identifie la resource ; 2. on charge la resource plus tard

~~~nit
var x = app.get_audio("x.wav")
assert x.is_loaded == false
x.load # charge le son
assert x.is_loaded == true
~~~

Le chargement pourait etre fait automatiquement lors de demande de service

~~~nit
var x = app.get_audio("x.wav")
assert x.is_loaded == false
x.play # charge implicitement puis joue
assert x.is_loaded == true
~~~

`app` devrait aussi pouvoir connaitres ses ressources, donc on pourait lui demander de charger des trucs à un moment donné pour éviter d'avoir des lags du au paresseux

~~~nit
var x = app.get_audio("x.wav")
assert x.is_loaded == false
app.load_all
assert x.is_loaded == true
x.play # direct play
~~~